tapir/shifts/forms.py

```diff
@@ -192,3 +192,82 @@ class Meta:
             "start_date": DateInput(),
             "end_date": DateInput(),
         }
+
+    confirm_cancelled_attendances = BooleanField(
+        label=_(
+            "I have read the warning about the cancelled attendances and confirm that the exemption should be created"
+        ),
+        required=False,
+        widget=HiddenInput,
+    )
+    confirm_cancelled_abcd_attendances = BooleanField(
+        label=_(
+            "I have read the warning about the cancelled ABCD attendances and confirm that the exemption should be "
+            "created "
+        ),
+        required=False,
+        widget=HiddenInput,
+    )
+
+    def validate_unique(self):
```

Comment:
Does this belong in validate_unique? Not quite sure what this does, but shouldn't there be a `clean` method or something along this lines that fits better semantically?

Reply:
I just had a quick look at the Django doc, I'm not quite sure when to use which. 
It works with both so I changed it to clean()